src/os/bluestore/BlueStore.cc

```diff
@@ -10305,8 +10305,6 @@ int BlueStore::_do_write(
   o->extent_map.compress_extent_map(dirty_start, dirty_end - dirty_start);
   o->extent_map.dirty_range(dirty_start, dirty_end - dirty_start);
 
-  r = 0;
-
  out:
```

Comment:
I would rather leave this in place.  Otherwise the code is more fragile because any change in r above this point will break the function return value.